### ISSUE TITLE
Infinispan cache does not expire

### ISSUE THREAD

**Issue by kdejaeger** [jhipster/generator-jhipster#9172]
I have used the current spring 2.1 branch to configure infinispan for my project. The configuration is almost the same. Just different entities and cache names. I cannot seem to figure out however why this caches doesn't expire. I thought to get support here first, since the configuration is almost identical and this issue might apply for jhipster as well. The cache defined in [this zip file](https://github.com/jhipster/generator-jhipster/files/2821456/cache.expiration.issue.zip) should expire in 15 minutes. (more details in zip file). It's similar to the spring local cache "local-app-data" as defined in CacheConfiguration.java.ejs, which is defined but never actually used in the standard generated project.

`        local_100k_15_m = new ConfigurationBuilder()
                .jmxStatistics().enabled(true)
                .clustering().cacheMode(LOCAL)
                .memory().evictionStrategy(REMOVE).evictionType(COUNT).size(100 * 1000).expiration().lifespan(15, MINUTES)
                .build();`

In reality it only expires when I clean the cache through JMX console. I have tried to only set the lifespan without the strategy/type/count and also tried adding maxIdle next to lifespan. But the cache never expires.

In JMX everything seems to appear to be using the same cache manager. Spring boot actuator lists all the caches in the system as well.

The only maybe weird thing I noticed is when I enable TRACE logs for org.infinispan, I only see trace activity around the hibernate caches, which are defined in the same way.

@srinivasa-vasu , any clue how I could figure this out?

**Comment by srinivasa-vasu** [comment]
@kdejaeger - This is from Infinispan documentation,

> Multiple cache operations such as put() can take a lifespan as parameter which defines the time when the entry should be expired. If you have no eviction configured and and you let this time expire, it can look as Infinispan has not removed the entry. For example, the JMX stats such as number of entries might not updated or the persistent store associated with Infinispan might still contain the entry. To understand what’s happening, it’s important to note that Infinispan has marked the entry as expired but has not actually removed it. Removal of expired entries happens in one of 2 ways: You try and do a get() or containsKey() for that entry.  The entry is then detected as expired and is removed. You have enabled eviction and an eviction thread wakes up periodically and purges expired entries.
If you have not enabled (2), or your eviction thread wakeup interval is large and you probe jconsole before the eviction thread kicks in, you will still see the expired entry. You can be assured that if you tried to retrieve the entry via a get() or containsKey() though, you won’t see the entry (and the entry will be removed).

Pls try the above suggested options instead of probing through jconsole.

**Comment by kdejaeger** [comment]
The way I know it's not working, is by putting a breakpoint in the cached method. As soon as evict the cache manually, either though the JMX console or through some Java code (cm.getCache(...).clear()), the breakpoint hits. Also Spring is logging the calculation of the cached 'key' and saying it found the entry.

**Comment by kdejaeger** [comment]
I added a @Listener (like [this blog post](https://www.baeldung.com/infinispan)) and see this when the cache entry is created:
`event = {EventImpl@22539} "EventImpl{type=CACHE_ENTRY_CREATED, pre=false, cache=Cache 'actieveDossiers'@5acc4eaa, key=[Organization[id=5,name=AZC Heerlen,organizationType=LOCATION], Organization[id=7,name=Leslocatie 1,organizationType=TRAINING_FACILITY]], value=[1], oldValue=null, transaction=null, originLocal=true, transactionSuccessful=false, entries=null, created=false, isCurrentState=false, available=false}"
 cache = {EncoderCache@22572}  size = 1
 key = {HashSet@22569}  size = 2
  0 = {Organization@22610} "Organization[id=5,name=AZC Heerlen,organizationType=LOCATION]"
  1 = {Organization@22611} "Organization[id=7,name=Leslocatie 1,organizationType=TRAINING_FACILITY]"
 metadata = {MetaParamsInternalMetadata@22598} "MetaParamsInternalMetadata{params=MetaParams{length=1, metas=[MetaLifespan=60000]}}"
 type = {Event$Type@22599} "CACHE_ENTRY_CREATED"
 value = {ArrayList@22600}  size = 1
  0 = {Long@22615} 1
   value = 1`

But is does not print any of the other annotations. Like cache visited.

**Comment by srinivasa-vasu** [comment]
@kdejaeger I tried the jhipster application 2.1.x branch. I tried through Spring abstraction local cache (local-app-data) for a 5ms lifespan. It's behaving as expected. Every time when I accessed same cache (after letting to expire 5ms+), it retrieved from the actual source. I didn't add the listeners per say as I could verify it through code flow as well as through jconsole. I am not sure what could be the reason why its not behaving as expected for you. Might have to look into the source. As you are not using jhipster app, why don't you check with Infinispan team?
